### PR TITLE
Adding database compatibility section

### PR DESCRIPTION
**Pull request by daviddavis**
[noissue]

### CODE REVIEW

File: docs/plugin-writer/planning-guide.rst [hunk 1]
```diff
@@ -176,3 +176,22 @@ as a publishing requirement.
 .. note::
    Write down any requirements for a webserver to interact with a client to facilitate content
    discovery.
+
+
+Database Compatibility
+^^^^^^^^^^^^^^^^^^^^^^
+
+Pulp attempts to support all the relational databases that django supports. However, plugin writers
```
Comment (dralley) [inline docs/plugin-writer/planning-guide.rst]: I kind of think that even this is too strong a claim.  We know sqlite is thoroughly broken w/ Pulp, and Django also supports e.g. Oracle DB which we have no intention of testing.  Maybe let's just say that we attempt to support Postgres and MySQL/MariaDB.

Comment (bmbouter) [reply docs/plugin-writer/planning-guide.rst]: Could we say something like:  "Pulp uses Django and theoretically could run on any relational database Django supports. In practice, Pulp formally supports PostgreSQL and MySQL/MariaDB, and is continuously tested against those."

I like identifying that Pulp is on top of Django and that it's possible-with-effort. I also like being more specific about what is tested and formally supported. When users file database compatability issues the formally supported aspect is what guides if the ticket is accepted/fixed or closed as wontfix.

Comment (bmbouter) [reply docs/plugin-writer/planning-guide.rst]: Also s/django/Django/g

Comment (daviddavis) [reply docs/plugin-writer/planning-guide.rst]: I'm a bit hesitant to make claims about which databases Pulp supports in this section--it wasn't really intended for that. I think maybe we should just change the name of this section to "Database Gotchas" and remove this first sentence. Thoughts?

**Comment by daviddavis** [comment]
@bmbouter @dralley I updated this PR to just reflect a list of things that might not be database compatible. I think we should probably break out our policy around which databases we support into a separate section. Let me know if you all disagree though.

**Comment by bmbouter** [comment]
@daviddavis that sounds good to me. @dralley is this ok w/ you also?

**Pull request merged**